src/ai/defaultai.cc

```diff
@@ -2926,6 +2977,17 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					number_of_same_nearby += same_it->second;
 				}
 
+				// Considering distance to a warehouse NOCOM
+				if (!bo.requires_supporters && !bo.inputs.empty()) {
```

Comment:
rather then doing this generally, I would prefer doing this for every subtype of building in the following.

Reply:
Generally I do not mind but we have many buildings x tribes, some of them not invented yet...

Reply:
what I meant is to think for each ai defined type of building (i.e. kRanger, kFisher, and so on) this would and should cover all buildings of all tribes and makes for the opportunity to have even scaled appliance of the malus.

Reply:
"scaled malus" - I would not do this now. Let training come up with some optimum for everyone

I think the malus should not be applied to some types (e.g. quary, lumberjacts, supporting sites). Remember it started as a request for placing warehouses. So this here is afterthought, I would not use it very extensively

Reply:
the request was about building productionsites near warehouses to shorten ware transport routes.  
And I agree not to (fully) apply it to some types or even give them a bonus. therefore i would have applied it (maybe scaled) seperately for each type of building.

Reply:
sorry, you are right
In fact if you put your ideas "on paper" I will transfer them to the code. I just believe current setup is quite good. There will be new commit soon, you can wait for it

Reply:
ok I made my mind. 
the general aplliance covers almost everything. 3 exceptions.
1. rangers are considered twice as they space consumer without inputs and not supporting producers
2. hunters should be applied a bonus if far away
3. frisian farms (i.e. supporting producers AND spaceconsumer) should be applied a bonus if farer away

Reply:
OK, I will prepare new commit

Reply:
Spaceconsumers generally should be positioned farer. Especially independent ones - regular barbiarian farm e.g.

Reply:
All productionsites are now handled on one place. What I implemented is not exactly what you wanted, but we will discuss